internal/thelma/app/builder/builder.go

```diff
@@ -156,9 +172,42 @@ func (b *thelmaBuilder) buildStateLoader(cfg config.Config, shellRunner shell.Ru
 		return statefixtures.NewFakeStateLoader(b.stateFixture.name, b.stateFixture.t, cfg.Home(), shellRunner)
 	}
 
-	sb, err := statebucket.New(cfg, clients.Google())
+	stateLoaderType, err := getStateLoaderType(cfg)
 	if err != nil {
 		return nil, err
 	}
-	return gitops.NewStateLoader(cfg.Home(), shellRunner, sb), nil
+
+	if stateLoaderType == gitopsStateLoader {
+		sb, err := statebucket.New(cfg, clients.Google())
+		if err != nil {
+			return nil, err
+		}
+		return gitops.NewStateLoader(cfg.Home(), shellRunner, sb), nil
+	} else if stateLoaderType == sherlockStateLoader {
+		sherlock, err := clients.Sherlock()
+		if err != nil {
+			return nil, err
+		}
+		return sherlockState.NewStateLoader(cfg.Home(), shellRunner, sherlock), nil
+	}
+
+	return nil, fmt.Errorf("received an undefined state loader source: valid options gitops or sherlock")
+}
+
+func getStateLoaderType(cfg config.Config) (stateLoaderType, error) {
+	var stateLoaderConfig StateLoaderConfig
+	if err := cfg.Unmarshal(stateLoaderConfigPrefix, &stateLoaderConfig); err != nil {
+		return 0, err
+	}
+
+	log.Info().Msgf("State Source: %s", stateLoaderConfig.Source)
+
+	switch stateLoaderConfig.Source {
+	case "gitops":
+		return gitopsStateLoader, nil
+	case "sherlock":
+		return sherlockStateLoader, nil
+	default:
+		return undefinedStateLoader, nil
```

Comment:
return error here instead of special undefined type? 🤔

Reply:
It does get bubbled up as an error from line 194 in this same file